File: day1/puzzle1/day1p1.py
file = open('input.txt', ('r+'), encoding=('utf-8'))

lines = list()
numArr = list()

for line in file:
    lines.append(line)

# Convert to list of individual ints
for num in line:
    numArr.append(int(num))

# Index and accumulator
counter = 0
total = 0

# Process array
for num in numArr:
    if counter < len(numArr) - 1:
        if num == numArr[counter + 1]:
            total += num
    counter += 1

# Check for circular case
if numArr[-1] == numArr[0]:
    total += numArr[-1]

print(f'This is the total: {total}')
file.close()

File: day1/puzzle1/day1p1.js
const fs = require('fs')

// Read file in, split on newline and pass the string to be processed
fs.readFile('./input.txt', (err, data) => {
    text = data.toString('utf-8').split('\n');
    console.log(`This is the total: ${process(text)}`);
});

// Processes the array
const process = (text) => {
    let charArr = text[0];
    let numArr = []

    // Convert to int
    for(let i = 0; i < charArr.length; i++){
        numArr[i] = parseInt(charArr[i]);
    }

    // Process the array
    let total = 0;
    for(let i = 0; i < numArr.length - 1; i ++){
        if (numArr[i] === numArr[i + 1]){
            total += numArr[i];
        }
    }

    // Check for circular case
    if(numArr[numArr.length - 1] === numArr[0]){
        total += numArr[numArr.length -1];
    }

    return total;
};